codechecker_lib/session_manager.py

```diff
@@ -0,0 +1,299 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+'''
+Handles the allocation and destruction of privileged sessions associated
+with a particular CodeChecker server.
+'''
+
+import os
+import shutil
+import uuid
+import json
+import hashlib
+import time
+import ldap
```

Comment:
importing ldap modules could be done optionally and if importing fails only the ldap authentication method will be disabled.
```
try:
   import ldap
   import ldap.sasl
except ImportError:
    # no ldap authentication is available
```